### PR TITLE
Added dark-mode availability as a user setting using a theme service

### PR DESCRIPTION
**Pull request by peetritense**
Added dark-mode availability as a user setting using a theme service and also combined with story 106009 for the splitting of files from carbon.scss

### CODE REVIEW

File: projects/valtimo/components/assets/css/carbon.scss [hunk 1]
```diff
@@ -14,19 +14,20 @@
...
+$valtimo-color-100: #002547;
+$valtimo-color-90: #002c54;
+$valtimo-color-80: #003361;
+$valtimo-color-70: #286198;
+$valtimo-color-60: #2b79bd;
+$valtimo-color-50: #61aedf;
+$valtimo-color-40: #8acff2;
+$valtimo-color-30: #aadcf6;
+$valtimo-color-20: #c9e9f9;
+$valtimo-color-10: #e9f6fd;
```
Comment (teodor-ritense) [inline projects/valtimo/components/assets/css/carbon.scss]: Why do you still need these if you have them in the colors.scss?

File: projects/valtimo/components/src/lib/components/right-sidebar/right-sidebar.component.ts [hunk 2]
```diff
@@ -136,14 +159,19 @@ export class RightSidebarComponent implements OnInit, OnDestroy {
...
-        this.showValtimoVersions = false;
-      }
-    });
+    this._subsciptions.add(
+      this.isAdmin$.subscribe(isAdmin => {
+        if (this.hideValtimoVersionsForNonAdmins && !isAdmin) {
+          this.showValtimoVersions = false;
+        }
+      })
+    );
```
Comment (teodor-ritense) [inline projects/valtimo/components/src/lib/components/right-sidebar/right-sidebar.component.ts]: You can remove this and the showValtimoVersions variable, it's not used anywhere as far as I can see

File: projects/valtimo/components/src/lib/components/right-sidebar/right-sidebar.component.ts [hunk 3]
```diff
@@ -126,6 +128,27 @@ export class RightSidebarComponent implements OnInit, OnDestroy {
...
+        value: SelectableCarbonTheme.G90,
+      },
+      {
+        name: this.translate.instant('settings.interface.themes.system'),
+        value: SelectableCarbonTheme.SYSTEM,
+      },
+    ])
+  );
+
+  private readonly _subsciptions = new Subscription();
```
Comment (teodor-ritense) [inline projects/valtimo/components/src/lib/components/right-sidebar/right-sidebar.component.ts]: Add the subscription in openFormSubscription() in there

File: projects/valtimo/components/src/lib/services/cds-theme.service.ts [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    this.openDarkModeSubscription();
+  }
+
+  public ngOnDestroy(): void {
+    this._subscriptions.unsubscribe();
+  }
+
+  public setPreferredTheme(selectedTheme: SelectableCarbonTheme): void {
+    if (selectedTheme) this._preferredTheme$.next(selectedTheme);
+  }
```
Comment (teodor-ritense) [inline projects/valtimo/components/src/lib/services/cds-theme.service.ts:46]:
```suggestion
  public setPreferredTheme(selectedTheme: SelectableCarbonTheme | undefined): void {
    if (selectedTheme) {
        this._preferredTheme$.next(selectedTheme);
     }
  }
```

File: projects/valtimo/components/src/lib/components/right-sidebar/right-sidebar.component.ts (hunk 2 above)
Comment (mbritense) [reply projects/valtimo/components/src/lib/components/right-sidebar/right-sidebar.component.ts]: This cannot be removed, it checks asynchronously if user is admin, hence variable name  hideValtimoVersionsForNonAdmins

File: projects/valtimo/components/assets/css/carbon.scss (hunk 1 above)
Comment (peetritense) [reply projects/valtimo/components/assets/css/carbon.scss]: Some components use the Valtimo-blue colors, like the blue tags.

File: projects/valtimo/components/src/lib/services/cds-theme.service.ts (hunk 4 above)
Comment (mbritense) [reply projects/valtimo/components/src/lib/services/cds-theme.service.ts:46]: I think the former looks cleaner, plus we have no style guide preferring one or the other

File: projects/valtimo/components/assets/css/carbon.scss [hunk 5]
```diff
@@ -14,19 +14,20 @@
...
 
 @use '@carbon/styles/scss/themes';
 @use '@carbon/colors';
+@use 'projects/valtimo/components/assets/css/color';
```
Comment (mbritense) [inline projects/valtimo/components/assets/css/carbon.scss]: @teodor-ritense is this correct?

**Review by teodor-ritense — DISMISSED** [review]

File: projects/valtimo/components/assets/css/carbon.scss (hunk 5 above)
Comment (teodor-ritense) [reply projects/valtimo/components/assets/css/carbon.scss]: No, it should be the relative path, @peetritense please fix

Comment (peetritense) [reply projects/valtimo/components/assets/css/carbon.scss]: Fixed

**Review by teodor-ritense — DISMISSED** [review]

**Review by mbritense — APPROVED** [review]

**Review by teodor-ritense — APPROVED** [review]

**Pull request merged**